### ISSUE TITLE
Can't pull images after Upgrading: Docker Engine 1.6.2 to Docker Engine 1.7.0 on RHEL 7 

### ISSUE THREAD

**Issue by marcellodesales** [docker/docker#14227]
Hi there,

I just got a new machine and decided to upgrade docker to the latest today... 

Problem
-------- 

I did a simple smoke test to pull the busybox and it failed with `Error pulling image (latest) from busybox, operation not supported`, after upgrading from Docker Engine 1.6.2 to Docker Engine 1.7.0. 

One important point here is that I had other 2 images only in this host.

```
$ docker pull busybox:latest
latest: Pulling from busybox
cf2616975b4a: Pull complete
cf2616975b4a: Download complete
6ce2e90b0bc7: Error downloading dependent layers
8c2e06607696: Error pulling image (latest) from busybox, operation not supported
Error pulling image (latest) from busybox, operation not supported
```

The errors in the `/var/log/messages` were as follows:

```
Jun 27 09:17:51 pppdc9prd8xp docker: time="2015-06-27T09:17:51.639886999Z" level=info msg="GET /v1.19/images/json"
Jun 27 09:17:59 pppdc9prd8xp docker: time="2015-06-27T09:17:59.324933444Z" level=info msg="DELETE /v1.19/images/cf2616975b4a"
Jun 27 09:18:03 pppdc9prd8xp docker: time="2015-06-27T09:18:03.264467934Z" level=info msg="GET /v1.19/images/json"
Jun 27 09:18:15 pppdc9prd8xp docker: time="2015-06-27T09:18:15.899904683Z" level=info msg="POST /v1.19/images/create?fromImage=busybox%3Alatest"
Jun 27 09:18:17 pppdc9prd8xp docker: time="2015-06-27T09:18:17.910857862Z" level=info msg="Image manifest for busybox:latest has been verified"
Jun 27 09:18:20 pppdc9prd8xp docker: time="2015-06-27T09:18:20.163018106Z" level=error msg="Error from V2 registry: operation not supported"
```

Downgrading it to 1.6.2 again made it to work...

```
# root at pppdc9prd8xp in /export/home/mdesales [14:51:57]
$ docker --version
Docker version 1.6.2, build 7c8fca2

# root at pppdc9prd8xp in /export/home/mdesales [14:56:38]
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
busybox             latest              8c2e06607696        10 weeks ago        2.43 MB

Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=info msg="-job create(coreos-registry-mysql-data) = ERR (1)"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=error msg="Handler for POST /containers/create returned error: No such image: busybox (tag: latest)"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=error msg="HTTP Error: statusCode=404 No such image: busybox (tag: latest)"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=info msg="POST /v1.18/images/create?fromImage=busybox&tag=latest"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=info msg="+job pull(busybox, latest)"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=info msg="+job resolve_repository(busybox)"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=info msg="-job resolve_repository(busybox) = OK (0)"
Jun 27 09:35:52 pppdc9prd8xp docker: time="2015-06-27T09:35:52Z" level=info msg="+job trust_update_base()"
Jun 27 09:35:53 pppdc9prd8xp docker: time="2015-06-27T09:35:53Z" level=info msg="-job trust_update_base() = OK (0)"
Jun 27 09:35:55 pppdc9prd8xp docker: time="2015-06-27T09:35:55Z" level=info msg="+job trust_key_check(/library/busybox)"
Jun 27 09:35:55 pppdc9prd8xp docker: time="2015-06-27T09:35:55Z" level=info msg="-job trust_key_check(/library/busybox) = OK (0)"
Jun 27 09:35:55 pppdc9prd8xp docker: time="2015-06-27T09:35:55Z" level=info msg="+job trust_key_check(/library/busybox)"
```

$ uname -a and OS
--------------

```
$ uname -a
Linux pppdc9prd8xp 3.10.0-229.4.2.el7.x86_64 #1 SMP Fri Apr 24 15:26:38 EDT 2015 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.1 (Maipo)
```

$ docker --version
---------
```
$ docker --version
Docker version 1.7.0, build 0baf609
```

$ docker info
----------------
```
$ docker -D info
Containers: 0
Images: 1
Storage Driver: overlay
 Backing Filesystem: nfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.4.2.el7.x86_64
Operating System: Red Hat Enterprise Linux
CPUs: 2
Total Memory: 7.641 GiB
Name: pppdc9prd8xp
ID: 3M2F:QYY7:Z5DI:YTVI:RAV4:SHPM:C3RC:CWIY:FHFA:ZYAS:SNHG:CMTY
```

$ docker version
-----------

```
$ docker version
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64
```

$ https://raw.githubusercontent.com/docker/docker/master/contrib/check-config.sh
-----

```
$ wget https://raw.githubusercontent.com/docker/docker/master/contrib/check-config.sh
$ ./check-config.sh
warning: /proc/config.gz does not exist, searching other paths for kernel config...
info: reading kernel config from /boot/config-3.10.0-229.4.2.el7.x86_64 ...

Generally Necessary:
- cgroup hierarchy: properly mounted [/sys/fs/cgroup]
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_DEVPTS_MULTIPLE_INSTANCES: enabled
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MACVLAN: enabled (as module)
- CONFIG_VETH: enabled (as module)
- CONFIG_BRIDGE: enabled (as module)
- CONFIG_BRIDGE_NETFILTER: enabled
- CONFIG_NF_NAT_IPV4: enabled (as module)
- CONFIG_IP_NF_FILTER: enabled (as module)
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled (as module)
- CONFIG_NF_NAT: enabled (as module)
- CONFIG_NF_NAT_NEEDED: enabled
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_MEMCG_SWAP: enabled
- CONFIG_MEMCG_SWAP_ENABLED: enabled
- CONFIG_RESOURCE_COUNTERS: enabled
- CONFIG_BLK_CGROUP: enabled
- CONFIG_IOSCHED_CFQ: enabled
- CONFIG_CGROUP_PERF: enabled
- CONFIG_CFS_BANDWIDTH: enabled
- Storage Drivers:
  - "aufs":
    - CONFIG_AUFS_FS: missing
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "btrfs":
    - CONFIG_BTRFS_FS: enabled (as module)
  - "devicemapper":
    - CONFIG_BLK_DEV_DM: enabled (as module)
    - CONFIG_DM_THIN_PROVISIONING: enabled (as module)
    - CONFIG_EXT4_FS: enabled (as module)
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "overlay":
    - CONFIG_OVERLAY_FS: enabled (as module)
    - CONFIG_EXT4_FS_SECURITY: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
  - "zfs":
    - /dev/zfs: missing
    - zfs command: missing
    - zpool command: missing
```

systemctl status docker
----------

```
$ systemctl status docker
docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled)
  Drop-In: /etc/systemd/system/docker.service.d
           └─http-proxy.conf
   Active: active (running) since Sat 2015-06-27 09:35:13 UTC; 5h 16min ago
     Docs: http://docs.docker.io
 Main PID: 4470 (docker)
   CGroup: /system.slice/docker.service
           └─4470 /usr/bin/docker -d -b=bridge0

Jun 27 09:35:58 pppdc9prd8xp docker[4470]: time="2015-06-27T09:35:58Z" level=info msg="+job log(start, 6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74, busybox:latest)"
Jun 27 09:35:58 pppdc9prd8xp docker[4470]: time="2015-06-27T09:35:58Z" level=info msg="-job log(start, 6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74, busybox:latest) = OK (0)"
Jun 27 09:35:58 pppdc9prd8xp docker[4470]: time="2015-06-27T09:35:58Z" level=info msg="-job start(6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74) = OK (0)"
Jun 27 09:35:58 pppdc9prd8xp docker[4470]: time="2015-06-27T09:35:58Z" level=info msg="POST /v1.18/containers/6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74/resize?h=24&w=204"
Jun 27 09:35:58 pppdc9prd8xp docker[4470]: time="2015-06-27T09:35:58Z" level=info msg="+job resize(6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74, 24, 204)"
Jun 27 09:35:58 pppdc9prd8xp docker[4470]: time="2015-06-27T09:35:58Z" level=info msg="-job resize(6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74, 24, 204) = OK (0)"
Jun 27 10:07:13 pppdc9prd8xp docker[4470]: time="2015-06-27T10:07:13Z" level=info msg="-job attach(6b402e093007cfe935a636b76c3da1c0df61b53d138a26a89b1fe04253016d74) = OK (0)"
Jun 27 14:42:31 pppdc9prd8xp docker[4470]: time="2015-06-27T14:42:31Z" level=info msg="GET /v1.18/containers/json"
Jun 27 14:42:31 pppdc9prd8xp docker[4470]: time="2015-06-27T14:42:31Z" level=info msg="+job containers()"
Jun 27 14:42:31 pppdc9prd8xp docker[4470]: time="2015-06-27T14:42:31Z" level=info msg="-job containers() = OK (0)"
```

Thanks for any help!
Marcello

**Comment by ankushagarwal** [comment]
Does this happen only after upgrade or even if you do a fresh install of 1.7 on a RHEL 7?

**Comment by marcellodesales** [comment]
@ankushagarwal not sure what the difference is, but I just downloaded 1.7, replaced the symlinks for the "docker" bin and started the daemon again... 

Is there anything else I would be supposed?

**Comment by ankushagarwal** [comment]
There was this thread a while ago about RHEL 7 adding some custom options (regards to the registry) to the docker binary : https://groups.google.com/forum/?fromgroups#!searchin/docker-user/Docker$20on$20RHEL$207/docker-user/uBs2__ELFe0/e8J6dNHvLr0J

Can you post the contents of `/etc/sysconfig/docker`

**Comment by ankushagarwal** [comment]
Can you also try 
```
docker pull docker.io/busybox:latest
```

**Comment by ankushagarwal** [comment]
Or
```
docker pull registry.hub.docker.com/busybox
```

**Comment by marcellodesales** [comment]
@ankushagarwal I could successfully build another machine and run 1.7 and the reason that I found is that the /var/lib/docker is mounted in a cloud disk over `nfs`... 

Based on the same base OS template I have, I got a problem regards the existence of the `devicemapper` directory... According to https://github.com/docker/docker/issues/14035#issuecomment-113533795, removing that would make it work... So, comparing the docker info between the broken host and the new one, I see that the Backing Filesystem are different and realized I had symlinked both servers differently (for storage size requirement differences)... 

So, I'm used to storing data in cloud disks and that's the first time I see this behavior... Is this a bug? Is this expected?

Working: Backing Filesystem = `extfs`
-------
```
# root at pppdc9prd8wu in /export/home/mdesales [6:58:50]
$ docker info
Containers: 0
Images: 3
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.4.2.el7.x86_64
Operating System: Red Hat Enterprise Linux
CPUs: 2
Total Memory: 7.641 GiB
Name: pppdc9prd8wu
ID: 3M2F:QYY7:Z5DI:YTVI:RAV4:SHPM:C3RC:CWIY:FHFA:ZYAS:SNHG:CMTY

# root at pppdc9prd8wu in /export/home/mdesales [7:00:21]
$ ls -la /var/lib/docker
lrwxrwxrwx. 1 root root 19 Jun 10 21:14 /var/lib/docker -> /app/var-lib-docker

# root at pppdc9prd8wu in /export/home/mdesales [7:05:35]
$ df
Filesystem                         1K-blocks    Used Available Use% Mounted on
/dev/mapper/rhel-root               19605504 2017872  17587632  11% /
devtmpfs                             3996484       0   3996484   0% /dev
tmpfs                                4005888       0   4005888   0% /dev/shm
tmpfs                                4005888   16840   3989048   1% /run
tmpfs                                4005888       0   4005888   0% /sys/fs/cgroup
/dev/mapper/vg_appdata-lvm_appdata 103077152   97104  97737376   1% /app
/dev/sda1                             508588  208476    300112  41% /boot

# root at pppdc9prd8wu in /export/home/mdesales [7:05:40]
$ docker pull registry.hub.docker.com/busybox
Pulling repository registry.hub.docker.com/busybox
8c2e06607696: Download complete
cf2616975b4a: Download complete
6ce2e90b0bc7: Download complete
Status: Image is up to date for registry.hub.docker.com/busybox:latest

# root at pppdc9prd8wu in /export/home/mdesales [7:13:36]
$ docker pull docker.io/busybox:latest
latest: Pulling from busybox
cf2616975b4a: Already exists
6ce2e90b0bc7: Already exists
8c2e06607696: Already exists
Digest: sha256:38a203e1986cf79639cfb9b2e1d6e773de84002feea2d4eb006b52004ee8502d
Status: Image is up to date for busybox:latest
```

* Not working:  Backing Filesystem = `nfs`
------
```
# root at pppdc9prd8xp in /docker-lib [6:59:03]
$ docker info
Containers: 0
Images: 1
Storage Driver: overlay
 Backing Filesystem: nfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.4.2.el7.x86_64
Operating System: Red Hat Enterprise Linux
CPUs: 2
Total Memory: 7.641 GiB
Name: pppdc9prd8xp
ID: 3M2F:QYY7:Z5DI:YTVI:RAV4:SHPM:C3RC:CWIY:FHFA:ZYAS:SNHG:CMTY
# root at pppdc9prd8xp in /docker-lib [7:00:24]
$ ls -la /var/lib/docker
lrwxrwxrwx. 1 root root 15 Jun 28 06:50 /var/lib/docker -> /docker-lib/lib

# root at pppdc9prd8xp in /docker-lib [7:05:21]
$ df
Filesystem                                                                       1K-blocks    Used Available Use% Mounted on
/dev/mapper/rhel-root                                                             19605504 2067504  17538000  11% /
devtmpfs                                                                           3996484       0   3996484   0% /dev
tmpfs                                                                              4005888       0   4005888   0% /dev/shm
tmpfs                                                                              4005888   16876   3989012   1% /run
tmpfs                                                                              4005888       0   4005888   0% /sys/fs/cgroup
/dev/mapper/vg_appdata-lvm_appdata                                               103077152  766952  97067528   1% /app_local
/dev/sda1                                                                           508588  201692    306896  40% /boot
pnxxqydhz001ssrp01cl01-n2-203n.corp.intuit.net:/qydhz001ssrp01_Cloud9_1435385977 307200000 3300896 303899104   2% /docker-lib

# root at pppdc9prd8xp in /docker-lib [7:05:25]
$ docker pull docker.io/busybox:latest
latest: Pulling from busybox
6ce2e90b0bc7: Download complete
6ce2e90b0bc7: Error downloading dependent layers
8c2e06607696: Error pulling image (latest) from busybox, operation not supported
cf2616975b4a: Download complete
Error pulling image (latest) from busybox, operation not supported

# root at pppdc9prd8xp in /docker-lib [7:13:18]
$ docker pull registry.hub.docker.com/busybox
Pulling repository registry.hub.docker.com/busybox
8c2e06607696: Error pulling image (latest) from registry.hub.docker.com/busybox, operation not supported
cf2616975b4a: Download complete
6ce2e90b0bc7: Error downloading dependent layers
Error pulling image (latest) from registry.hub.docker.com/busybox, operation not supported
```

Here are the other stuff you requested, which are the same in both servers...

```
# root at pppdc9prd8wu in /export/home/mdesales [6:19:03]
$ cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.1 (Maipo)
```

cat /etc/sysconfig/docker
---------
```
# root at pppdc9prd8xp in /export/home/mdesales [6:05:31]
$ cat /etc/sysconfig/docker
# /etc/sysconfig/docker
#
# Other arguments to pass to the docker daemon process
# These will be parsed by the sysv initscript and appended
# to the arguments list passed to docker -d
other_args="-b=bridge0"
```

My bridge0 
---------
```
# root at pppdc9prd8wu in /export/home/mdesales [6:19:15]
$ ifconfig
bridge0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 192.168.5.1  netmask 255.255.252.0  broadcast 192.168.7.255
        ether 00:00:00:00:00:00  txqueuelen 0  (Ethernet)
        RX packets 18575  bytes 1866270 (1.7 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 10778  bytes 786346 (767.9 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

------

**Comment by marcellodesales** [comment]
@ankushagarwal, I will be closing this and if this is NOT a feature, we should open a new issue...
